docs/config.md

```diff
@@ -0,0 +1,42 @@
+# Configuration
+
+The CSI driver instance is configured by a combination of command-line flags and a YAML configuration
+file. You can use the following flags:
+
+- `--config`: path to the YAML configuration file, optional;
+- `--nodeid`: the current Kubernetes node identifier, must be the same than the node it's running on
+- `-—refreshresources`: toggles the operation mode, when enabled it continuously watch for `Secret`
```

Comment:
OK I'm struggling a bit with the presentation here, and the distinction of what is available in general from the command line, vs. what we envision a cluster admin wanting to manipulate.

I'm going to briefly present things in my words @otaviof and then let's see how we can reconcile that with what you are trying to convey here.

1) I envision initially surfacing via the config map the only the subset of command line options we think the cluster admin will want to manipulate / we think the *should* manipulate

I see 3 of those:
- share-rellist-interval
- refreshresources
- ignorenamespace

From a code reference perspective

https://github.com/openshift/csi-driver-shared-resource/blob/c8dab4fb90277123868abe8f05086fa48ed1f2d5/cmd/main.go#L86-L89

2) yes, we currently have the other 4 command line options at https://github.com/openshift/csi-driver-shared-resource/blob/c8dab4fb90277123868abe8f05086fa48ed1f2d5/cmd/main.go#L81-L85

- endpoint
- drivername
- nodeid
- maxvolumespernode

to be fair, it is not readily apparent to anyone but me, since I am the one who "originated" this driver, but these and config options that existing in the upstream, generic CSI hostpath/secret share volume drivers I originally based csi-driver-share-resource from.

most likely, as we move toward tech preview and GA, we'll either get rid of these entirely, or we'll keep them hidden, in reserve, in case user interaction and feedback informs us of some tuning for those we are currently not anticipating.

So, digest all this ^^ @otaviof, let's talk "voice to voice" if need be, and then let's see how we change this up

also to be fair, none of this nuance is captured in https://issues.redhat.com/browse/BUILD-340

Reply:
Thanks for the explanation ([here](https://github.com/openshift/csi-driver-shared-resource/pull/60#discussion_r729050639)) about the historical origins of those flags, it gives the input need so we can define a clear distinction of what goes where.

I'm following your suggestion, please consider latest changes.